src/messages/text_document_did_open.cc

```diff
@@ -62,9 +62,11 @@ struct TextDocumentDidOpenHandler
 
     // Submit new index request.
     const Project::Entry& entry = project->FindCompilationEntryForFile(path);
```

Comment:
Remove const to fix compile failure

Reply:
That doesn't work. `FindCompilationEntryForFile` returns a temporary object of type `Project::Entry` which can not be bound to a non-constant reference to `Project::Entry`.

Further more, changing `FindCompilationEntryForFile` returns a copy of a local variable of type `Project::Entry`, so simply changing it to return a reference will cause undefined behaviour as soon as we hit `return result;`.

I'm looking into it, but any help would be nice since I'm still fairly new to the cquery database.

Reply:
I can easily make it a unique pointer, but that doesn't solve #541, as it will be freed as soon as `TextDocumentDidOpenHandler()` returns.

Reply:
Looks like the existing code is buggy (it should not be a reference). I can do some refactoring to fix this more cleanly, ala TryGetCompilationEntryForFile() which takes a lambda which gets passed the Project::Entry* instance. Then FindCompilationEntryForFile becomes a wrapper around this function.

Reply:
Updated and this time it compiles. :man_facepalming: